src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/AzureServiceBus/AzureServiceBusWrapperBase.cs

```diff
@@ -0,0 +1,20 @@
+// Copyright 2020 New Relic, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using NewRelic.Agent.Api;
+using NewRelic.Agent.Extensions.Providers.Wrapper;
+
+namespace NewRelic.Providers.Wrapper.AzureServiceBus
+{
+    public abstract class AzureServiceBusWrapperBase : IWrapper
+    {
+        protected const string BrokerVendorName = "AzureServiceBus";
+
+        public bool IsTransactionRequired => true; // only instrument service bus methods if we're already in a transaction
```

Comment:
This seems different compared with other MQ that we instrument.  Shouldn't the receive generate a transaction?

Reply:
Probably a topic for pairing on Tuesday. But this implementation is consistent with our MSMQ instrumentation. Unlike RabbitMQ, there's not an "eventing" layer that we could instrument that would wrap processing time on the client application.

Reply:
Looks like we could potentially instrument `ServiceBusProcessor` which does implement an event-based receive model where client code is called when a message is received. I'll take a quick look to see if it's worth the effort or not.